tornado/speedups.c

```diff
@@ -14,13 +17,35 @@ static PyObject* websocket_mask(PyObject* self, PyObject* args) {
         return NULL;
     }
 
+    uint32_mask = ((uint32_t*)mask)[0];
+
     result = PyBytes_FromStringAndSize(NULL, data_len);
     if (!result) {
         return NULL;
     }
     buf = PyBytes_AsString(result);
+
+    if (sizeof(size_t) >= 8) {
+        uint64_mask = uint32_mask;
+        uint64_mask = (uint64_mask << 32) | uint32_mask;
+
+        while (data_len >= 8) {
+            ((uint64_t*)buf)[0] = ((uint64_t*)data)[0] ^ uint64_mask;
+            data += 8;
+            buf += 8;
+            data_len -= 8;
+        }
+    }
+
+    while (data_len >= 4) {
```

Comment:
another silly little thing which probably has no measurable effect: this can just be `if (data_len >= 4)` because it can only be true once, saving an extra compare/branch after the first (only) iteration